### DISCUSSION TITLE
Bind telescope "symbols" to the "s" key

### DISCUSSION THREAD

**Discussion by vdegenne** [LazyVim/LazyVim#4422] [Q&A] [answered]
Pressing s key in normal mode is bringing flash search, however I already have flash on my / key.
That means I have a free s key that I would want to use to open current buffer symbols using telescope which is natively bind to `<leader>ss` (which I find exhausting over time, ring finger is not the most precise finger).

I am still a noob when it comes to configuration, I just know I have 2 solutions: create a keymap in `keymaps.lua` or customize telescope from a file in `plugins` directory (which I conveniently already have.)

What method would you recommend? If you could show me the code to achieve that, that would be even better!

Thanks for the help in advance.

**Comment by dpetka2001** [comment, marked as answer]
```lua
return {
  {
    "nvim-telescope/telescope.nvim",
    keys = {
      {
        "s",
        function()
          require("telescope.builtin").lsp_document_symbols({
            symbols = LazyVim.config.get_kind_filter(),
          })
        end,
        desc = "Goto Symbol",
      },
    },
  },
  {
    "folke/flash.nvim",
    keys = {
      { "s", mode = { "n", "x", "o" }, false },
    },
  },
}
```

**Reply by vdegenne** [reply]
Thanks a lot it works!
By the way just as a parenthesis, recently I have an issue when telescope symbols window open It writes an A out of nowhere. And the problem persists with the new "s" key binding. I don't think I have anything in my settings causing that. Is that a known issue?

**Reply by vdegenne** [reply]
https://github.com/user-attachments/assets/aebade39-4059-42ae-96e7-32d84620cc98

**Reply by dpetka2001** [reply]
You have 2 typescript lsp servers attached on the same buffer. Disable one and i think you won't see any `A`.

**Reply by vdegenne** [reply]
I noticed search results were doubled in telescope just before reading your comment, I don't know how you knew about that but it sounds about right. Sorry to take opportunity of the thread, could you just tell me why I have 2 lsp servers running and how I can prevent that? (The issue persists everywhere, not just in this project.) Thanks.

**Reply by dpetka2001** [reply]
Well lazyvim only has vtsls server for typescript. So you probably enabled ts_ls (former tsserver) on your own. Just uninstall it from mason and make sure you don't have any ts_ls in your `nvim-lspconfig` spec. Or if you want to explicitly use ts_ls, just disable the Lazyvim typescript extra as they are not compatible. Your choice.

**Reply by vdegenne** [reply]
When using `LspInfo` I see `vtsls` and `ts_ls` clients. The weird thing is I don't know where they got activated from. Even if I disabled all extra plugins and my plugin directory, they both get activated when I open a ts file.

**Reply by dpetka2001** [reply]
Search in your configuration for `tsserver` or `ts_ls`. Lazyvim typescript extra only enables `vtsls` by default.

**Reply by vdegenne** [reply]
@dpetka2001 Ok thanks, it's back to normal. I didn't know Mason installs were persistent even after commenting typescript extra in the config. Just got little bit confused there.

**Reply by dpetka2001** [reply]
Yes Mason doesn't uninstall anything by default. So if a user disables something in his configuration he also has to manually uninstall corresponding lsp server from Mason UI as well.
